pkg/operator/controllers/genevalogging/genevalogging.go

```diff
@@ -68,7 +40,12 @@ func (g *genevaLogging) securityContextConstraints(ctx context.Context, name, se
 	return scc, nil
 }
 
-func (g *genevaLogging) daemonset(r azure.Resource) *appsv1.DaemonSet {
+func (g *GenevaloggingReconciler) daemonset(cluster *arov1alpha1.Cluster) (*appsv1.DaemonSet, error) {
+	r, err := azure.ParseResourceID(cluster.Spec.ResourceID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Old function used to use `r` to create daemonset, now we are using the `*arov1alpha1.Cluster` instead.  Is this necessary?

Reply:
Yes.